service/handlers.go

```diff
@@ -14,102 +14,119 @@ const (
 )
 
 func GetBookByID(c *gin.Context) {
-	updateUserData(c)
+	err := updateUserData(c)
+	if err != nil {
+		c.JSON(http.StatusInternalServerError, gin.H{"Error": "Failed To Update User Data"})
+	}
+
 	db := data_store.NewBookStorer()
 	if db == nil {
```

Comment:
check if the `db` nil when you are starting your service (because it's a factory)